.github/workflows/ci.yml

```diff
@@ -38,4 +38,17 @@ jobs:
       - name: Test
         run: cargo test
         env:
-          RUST_BACKTRACE: 1
\ No newline at end of file
+          RUST_BACKTRACE: 1
+  
+      - name: Compute diff between local and upstream candid files
+        run: |
+          curl -s https://raw.githubusercontent.com/dfinity/cdk-rs/main/src/ic-certified-assets/assets.did > upstream_assets.did
+          echo CANDID_DIFF=$(diff upstream_assets.did assets.did | wc -l) >> $GITHUB_ENV
+          
+      - name: Fail if local candid file different from upstream
+        if: ${{ env.CANDID_DIFF != 0 }}
+        uses: actions/github-script@v3
+        with:
+          script: |
+              core.setFailed('Candid file for asset canister is different from upstream (dfinity/cdk-rs/ic-certified-assets)')
+              
```

Comment:
I recommend to make the workflow report these in a status that does not depend on the matrix.  This allows us to set up branch protection rules that don't have to change when the rust version changes.

Example: https://github.com/dfinity/cdk-rs/blob/main/.github/workflows/e2e.yml#L55

Alternately, I'd suggest to make a separate workflow for this check, since it doesn't depend on rust and won't need a matrix at all.

Reply:
I wanted to avoid installing cargo twice, but you're right, that's the correct approach. 

I have this good-bad habit where I'll try to save computational resources to reduce carbon emission - a drop in the ocean, which makes no sense here.

Reply:
disregard what I said about cargo - my mind was in another GH Action yml at the time of writing...

Reply:
I moved this step into separate workflow `ci-candid.yml`